### PR TITLE
Remove obsolete upper bound check

### PR DESCRIPTION
**Pull request by jriedel-ionos**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This check is already done in https://github.com/kubernetes/autoscaler/blob/92c182feda0a08a7ff594bff17f457713294f61b/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go#L437-L447 so we don't have to do it again.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/ionoscloud/ionoscloud_cloud_provider.go [hunk 1]
```diff
@@ -76,9 +76,6 @@ func (n *nodePool) IncreaseSize(delta int) error {
 		return err
 	}
 	targetSize := size + delta
-	if targetSize > n.max {
-		return fmt.Errorf("size increase exceeds upper bound of %d", n.max)
-	}
```
Comment (piepmatz) [inline cluster-autoscaler/cloudprovider/ionoscloud/ionoscloud_cloud_provider.go:81]: I thought about this again. Removing this check is fine, but I guess it's not enough.

We saw this check fail when `targetSize == n.max + 1`.
Assuming that `n.max` and `delta` are alright, this means that `targetSize` is larger than it should be. Just removing this check here would just move the error to the Cloud API validations, as they would complain about exceeding the maximum.

We need to find out why `targetSize` has the wrong value. Here it's based on `GetNodeGroupSize()`. The included non-provider-specific size check uses something different: `TargetSize()` which in turn calls `GetNodeGroupTargetSize()`. They are different:
- `GetNodeGroupSize()` returns the current number of actually existing nodes in the node pool. 
- `GetNodeGroupTargetSize()` returns the node pool's specified (i.e. desired) node count.

As we're about to update the node pool's desired node count, we should use the latter. But we don't. I think that might be the root cause. So please also switch to `GetNodeGroupTargetSize()`.

**Review by piepmatz — CHANGES_REQUESTED** [review]

**Review by piepmatz — APPROVED** [review]

**Comment by Shubham82** [comment]
Giving `LGTM` (to merge) as the Owner of ionoscloud approved this PR.
/lgtm

CC @feiskyer PTAL!

**Comment by Shubham82** [comment]
Owners of ionoscloud cloud provider are not Kubernetes members, so they can't approve the PR. (like @piepmatz gave `LGTM` but it didn't apply).

As they reviewed the PR, so can we merge this PR?

@gjtempleton @MaciekPytel @x13n PTAL!

**Comment by Shubham82** [comment]
> @x13n / @Shubham82 will this be included in the next 1.31.x patch release?

IMO, we can cherry-pick these changes(as it is a bug fix) to the supported CA releases (CA 1.28 - CA 1.31).

cc @x13n @piepmatz WDYT?

**Comment by avorima** [comment]
@Shubham82 we're hoping to get the next 1.31 CA release to include this fix. any older versions would be nice as well

**Comment by Shubham82** [comment]
> @Shubham82 we're hoping to get the next 1.31 CA release to include this fix. any older versions would be nice as well

Yeah, if this bug exists in the previous CA releases(supported) then it will be good to fix it in those releases.

**Comment by x13n** [comment]
Cherrypicking makes sense to me.

/cherry-pick cluster-autoscaler-release-1.31
/cherry-pick cluster-autoscaler-release-1.30
/cherry-pick cluster-autoscaler-release-1.29

**Pull request merged**